src/Tools/dotnet-dump/Analyzer.cs

```diff
@@ -139,13 +117,19 @@ public async Task<int> Analyze(FileInfo dump_path, string[] command)
         private void AddServices(DataTarget target)
         {
             _serviceProvider.AddService(target);
+            _serviceProvider.AddService(target.DataReader);
             _serviceProvider.AddService<IConsoleService>(_consoleProvider);
             _serviceProvider.AddService(_commandProcessor);
             _serviceProvider.AddServiceFactory(typeof(IHelpBuilder), _commandProcessor.CreateHelpBuilder);
 
+            if (!(target.DataReader is IThreadReader threadReader))
+            {
+                throw new NotImplementedException("IThreadReader not implemented");
```

Comment:
I think NotSupportedException or InvalidOperationException would be more appropriate.  NotImplementedException is usually for when you hit a codepath within this tool which isn't yet implemented.  Usually you don't throw NIE on behalf of other code.